### ISSUE TITLE
`Logger.includes` broken in 2.8.x

### ISSUE THREAD

**Issue by rtar** [outr/scribe#129]
`Logger.includes` does not work properly in `2.8.x`. It used to work correctly in `2.7.13`.

I.e. the following expression will return `true` in `2.7.13` and will return `false` in `2.8.x`:
```
scribe.Logger.root includes scribe.Level.Info
```

The reason for that is the following change. There used to be a check if the level is included in this `Logger` OR the parent of this `Logger`. The logic was (accidentially?) changed to AND, i.e it requires _both_ this `Logger` to include the level and the parent. As this is a recursive logic, then this check goes to `root`, which does not have a parent and `parentId.map(Logger.apply).exists(_.includes(level))` always returns `false`.

See more here:
https://github.com/outr/scribe/commit/b4073506919b0a020f5ccbf2b4eb93d6cad85c40#diff-1457cb600ebaedcf2615c194e0435635R40

**Comment by darkfrog26** [comment]
Good catch. This will be fixed in the next release.

**Comment by darkfrog26** [comment]
Take a look at this update to `includes` to simplify the functionality and let me know if you think that's a reasonable fix: https://github.com/outr/scribe/commit/1b4b1f907a2a2f2131284197c47cc48e4d8e60e0#diff-1457cb600ebaedcf2615c194e0435635R39

**Comment by rtar** [comment]
Well, it works now, it seems. My original issue was that all slf4j code suddenly stopped working, because `ScribeLoggerAdapter` implements `isInfoEnabled` as `includes(Level.Info)`, and it always returned `false`. I did not have the time to check if slf4j works with new version properly, but I assume it should because `includes` now works properly, it seems.

As for the implementation, I would add a unit test somewhere into `LoggingSpec` like this because it seems to be an important functionality that is easy to break:
```
"confirm info level is included by default" in {
  assert(Logger.root.includes(Level.Info))
}
```

Also `asInstanceOf` looks a bit scary here, but, I am not familiar with Scribe codebase enough to say if it is evil, may be it is okay:
https://github.com/outr/scribe/commit/1b4b1f907a2a2f2131284197c47cc48e4d8e60e0#diff-1457cb600ebaedcf2615c194e0435635R50

Offtopic: while debugging the issue I have noticed that `getName` always returns `null` in `ScribeLoggerAdapter`. The reason for that is that `name: String` inside of `ScribeLoggerAdapter` does not override protected `name` field coming from `org.slf4j.helpers.NamedLoggerBase` (it would, probably, have been overriding it if it was `val name: String` instead). May be that was intended, just sharing the finding.

**Comment by darkfrog26** [comment]
I actually wrote a test like that last night, but it was passing with the existing code.

**Comment by darkfrog26** [comment]
Okay, I fixed the `getName` issue. As for the `asInstanceOf`, I don't love it either, but it's a short-term solution until I make some additional changes to the `LogModifier` typing.

Let me know if there are any other suggestions and I'll get a new release out today or tomorrow.

**Comment by rtar** [comment]
Sounds perfect to me, no suggestions.

**Comment by darkfrog26** [comment]
Resolved with 2.8.2 release
